src/components/App/App.js

```diff
@@ -1,24 +1,50 @@
-import React from 'react'
+import React, { Component } from 'react'
 import {
-  BrowserRouter as Router,
+  Router,
   Route,
-  Switch
 } from 'react-router-dom'
-import { Login, Signup, Home } from '../Components'
-import './App.scss'
+import { Board, Callback, Home, LoggedOut, Profile } from '../Components'
+import { Auth, History } from '../../services/Services'
 
-const App = () => {
-  return (
-    <div className="app-routes">
-      <Router>
-        <Switch>
-          <Route path="/login" component={Login} />
-          <Route path="/signup" component={Signup} />
-          <Route path="/" component={Home} />
-        </Switch>
-      </Router>
-    </div>
-  )
+const auth = new Auth()
+
+class App extends Component {
+  handleAuthentication = nextState => {
+    if (/access_token|id_token|error/.test(nextState.location.hash)) {
+      auth.handleAuthentication()
+    }
+  }
+
+  render() {
+    // calls the isAuthenticated method in authentication service
+    const { isAuthenticated } = auth
+    return (
+      <div>
+        <Router history={History}>
+          <div>
+            <Route exact path="/" component={Home} />
+            <Route path="/home" component={Home} />
+            {isAuthenticated() &&
+              <Route path="/board" component={Board} />
+            }
+            {
+              isAuthenticated() &&
```

Comment:
Why do you call `isAuthenticated` twice? 🤔
What do you think about wrapping both of them?
```jsx
{isAuthenticated() && (
	<Fragment>
		<Route .../>
		<Route .../>
	</Fragment>
)}

```

Reply:
There is no good reason. If you want you can go ahead and change it so it encapsulates all the protected routes.

Reply:
@nickeyvee I think you must change that. Because it is your code, after that change I will approve. 👍